fmskill/model/factory.py

```diff
@@ -42,42 +64,107 @@ class ModelResult:
+            pd.Series: ModelResult.from_pd_series,
+        }
+
+        return input_data_handler_mapping[type(data)](
+            data,
+            item,
+            *args,
+            **kwargs,
+        )
 
-    @staticmethod
-    def _mr_or_mr_item(mr):
-        if mr._selected_item is not None:
-            return mr[mr._selected_item]
+        # if isinstance(data, (pd.DataFrame, pd.Series)):
+        #     type_ = kwargs.pop("type", "point")
+        #     if type_ == "point":
+        #         mr = DataFramePointModelResult(data, *args, **kwargs)
+        #     elif type_ == "track":
+        #         mr = DataFrameTrackModelResult(data, *args, **kwargs)
+        #     else:
+        #         raise ValueError(f"type '{type_}' unknown (point, track)")
+        #     return self._mr_or_mr_item(mr)
+
+    @classmethod
+    def from_filepath(
+        cls,
+        filepath: Union[str, Path, list],
+        item: ItemSpecifier = None,
+        *args,
+        **kwargs,
+    ):
+        filename = _as_path(filepath)
+        ext = filename.suffix
+        if "dfs" in ext:
+            dfs: DfsType = mikeio.open(filename)
+            return ModelResult.from_dfs(dfs, item)
         else:
-            return mr
+            if "*" not in str(filename):
+                return ModelResult.from_xarray_dataset(
+                    xr.open_dataset(filename), item, *args, **kwargs
+                )
+            elif isinstance(filepath, str) or isinstance(filepath, list):
+                return ModelResult.from_xarray_dataset(
+                    xr.open_mfdataset(filepath), item, *args, **kwargs
+                )
+
+    @classmethod
+    def from_dfs(cls, dfs: DfsType, item: ItemSpecifier = None):
+        """Create a ModelResult from a dfs instance"""
+        idx = dfs_get_item_index(dfs, item)
+        return DfsModelResultItem(
+            dfs=dfs, itemInfo=dfs.items[idx], filename=None, item_index=idx
+        )
+
+    @classmethod
+    def from_xarray_array(cls, da: xr.DataArray, *args, **kwargs):
+        """
+        Create a ModelResult from an xarray DataArray. This should be the interface
+        used for creating single item ModelResults from any xarray data structure.
+        """
+        new_da = validate_and_format_xarray(da)
+        return XArrayModelResultItem(new_da, *args, **kwargs)
+
+    @classmethod
+    def from_xarray_dataset(
+        cls, ds: xr.Dataset, item: ItemSpecifier = None, *args, **kwargs
+    ):
+        """Create a ModelResult from an xarray Dataset. An item needs to be passed if
+        the Dataset contains more than one item."""
+
+        item = cls._xarray_get_item_name(ds, item)
+        return cls.from_xarray_array(ds[item], *args, **kwargs)
+
+    @classmethod
+    def from_mikeio_array(cls, da: mikeio.DataArray, *args, **kwargs):
+        """Create a ModelResult from a mikeio DataArray."""
+        return cls.from_xarray_array(da.to_xarray(), *args, **kwargs)
```

Comment:
Please note that we lose functionality by converting a mikeio.DataArray to xr.DataArray ! If it's a dfsu-type of DataArray we can no longer extract tracks if we convert to to xr!

Reply:
@jsmariegaard Yep, this is outdated code that is no longer in use :( I'll do a proper cleanup session soon to hopefully make the code review easier.